File: java17/src/main/java/com/simplify/java17/BookParserController.java
package com.simplify.java17;

import com.simplify.java17.book.BookParser;
import com.simplify.java17.book.BookResponse;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

import java.nio.file.Path;

@RestController
class BookParserController {

    private final BookParser bookParser;

    BookParserController(BookParser bookParser) {
        this.bookParser = bookParser;
    }

    @GetMapping("/csv-to-json")
    BookResponse convertCsvToJson(@RequestParam String inputCsv, @RequestParam String outputJson) {
        return new BookResponse(bookParser.convertCsvToJson(Path.of(inputCsv), Path.of(outputJson)), 1.0d);
    }

}

File: ktlin/src/main/kotlin/com/simplify/ktlin/BookParserController.kt
package com.simplify.ktlin

import com.simplify.ktlin.book.BookParser
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.bind.annotation.RestController
import java.io.IOException

data class BookResponse(val json: String, val version: Double)

@RestController
class BookParserController(private val bookParser: BookParser) {

    @GetMapping("/csv-to-json")
    fun convertCsvToJson(@RequestParam inputCsv: String, @RequestParam outputJson: String): BookResponse =
        BookResponse(bookParser.convertCsvToJson(inputCsv, outputJson), 1.0)
}